src/main/java/graphql/execution/instrumentation/dataloader/PerLevelDataLoaderDispatchStrategy.java

```diff
@@ -275,9 +312,84 @@ private boolean levelReady(int level) {
     }
 
     void dispatch(int level) {
-        DataLoaderRegistry dataLoaderRegistry = executionContext.getDataLoaderRegistry();
-        dataLoaderRegistry.dispatchAll();
+        // if we have any DataLoaderCFs => use new Algorithm
+        if (callStack.levelToDFEWithDataLoaderCF.get(level) != null) {
+            dispatchDLCFImpl(callStack.levelToDFEWithDataLoaderCF.get(level), true);
```

Comment:
The value is unlikely to change between invocations, but in theory, could. Perhaps the result of the lookup should be memoized and then null checked and passed into `dispatchDLCFImpl`